compose/compose-file/index.md

```diff
@@ -1562,6 +1655,35 @@ refer to it within the Compose file:
         external:
           name: actual-name-of-network
 
+## configs configuration reference
+
+The top-level `configs` declaration defines or references
+[configs](/engine/swarm/configs.md) which can be granted to the services in this
+stack. The source of the config is either `file` or `external`.
+
+- `file`: The config is created with the contents of the file at the specified
+  path.
+- `external`: If set to true, specifies that this config has already been
+  created. Docker will not attempt to create it, and if it does not exist, a
+  `config not found` error occurs.
+
+In this example, `my_first_config` will be created (as
+`<stack_name>_my_first_config)`when the stack is deployed,
+and `my_second_config` already exists in Docker.
+
+```none
+configs:
+  my_first_config:
+    file: ./config_data
+  my_second_config:
+    external: true
```

Comment:
It might be missing from the `secrets` doc as well (or intentionally omitted), but another valid syntax is
```
my_third_config:
  external:
    name: name_of_config_on_engine
```